frontend/src/components/RequestDialog.vue

```diff
@@ -117,14 +134,22 @@ const {
   error: requestSignerError,
 } = useRequestSigner();
 
-const runRequestSigner = () => {
+const connectMetamask = async () => {
   // TOOD: In future we will not separate getting provider and signer which
   // resolve the undefined provider case.
+  provider.value = await createMetaMaskProvider();
   if (provider.value) {
     requestSigner(provider.value);
+    settings.setConnectedWallet(WalletType.Metamask);
   }
 };
 
+const connectWalletConnect = async () => {
+  const rpcUrlsList = useRpcUrls(configuration.chains);
+  provider.value = await createWalletConnectProvider(rpcUrlsList);
+  settings.setConnectedWallet(WalletType.WalletConnect);
```

Comment:
createWalletConnectProvider may return undefined. I think in this case we should not call this method.

Reply:
yes, right.. will do!